src/WrapperApp/components/Results/EstimatorsSelect/EstimatorsSelect.tsx

```diff
@@ -28,21 +35,52 @@ const EstimatorsSelect = ({
 			const estimatorExists =
 				currentEstimatorData[id] && currentEstimatorData[id].name !== '';
 
-			if (!estimatorExists) {
-				const simulationJobId = simulation.jobId;
+			if (estimatorsPagesData) {
+				const estimatorsPagesByDimensions = estimatorsPagesData[id].pagesByDimensions;
+
+				if (!estimatorExists) {
+					const simulationJobId = simulation.jobId;
+					const allResults: SpecificEstimator[] = [];
+
+					for (const [dimension, pageDimension] of Object.entries(
+						estimatorsPagesByDimensions
+					)) {
+						const estimatorData = await getJobResult(
+							{
+								jobId: simulationJobId,
+								estimatorName: estimatorsTabData[id],
+								pageNumbers: pageDimension.pageNums
+							},
+							controller.signal,
+							true
+						);
+
+						if (estimatorData) {
+							allResults.push(estimatorData);
+						}
+					}
 
-				const estimatorData = await getJobResult(
-					{
-						jobId: simulationJobId,
-						estimatorName: estimatorsTabData[id]
-					},
-					controller.signal,
-					true
-				);
+					if (allResults.length === 0) return;
 
-				const newEstimatorData = estimatorData?.estimators;
+					const mergedResults = allResults.reduce(
```

Comment:
what kind of merging happens here ? in yaptide by merging we usually mean aggregation of data from many parallel processes. If you simply need to put together data from all pages, then maybe you should use word `concatenation` or `aggregation` ?

Reply:
There are merged results of responses of pages for one estimator, I may change the name to make more readable

Reply:
In the context of most operations merging = "split the input into 15 parallel runs, take each of the 15 files, read them into memory, average 15 arrays and thats the merged result". Therefore I suggest some renaming to avoid confusion and naming clash